qa/rpc-tests/mempool_size_limit.py

```diff
@@ -66,60 +73,128 @@ def load_from_storage(path):
 def satoshi_round(amount):
     return Decimal(amount).quantize(Decimal('0.00000001'), rounding=ROUND_DOWN)
 
-def create_tx_script():
-    script_pubkey = "6a4d0200" #OP_RETURN OP_PUSH2 512 bytes
-    for i in range (512):
-        script_pubkey = script_pubkey + "01"
-
-    ## concatenate 128 txouts of above script_pubkey which we'll insert before the txout for change
-    txouts = "80" # i.e. decimal 128
-    for k in range(128):
+def amount_to_satoshi(amount):
+    return int(Decimal(amount * 100000000).quantize(Decimal('1'), rounding=ROUND_05UP))
+
+def create_tx_script(script_size, amount):
+    bytes_per_out = script_size // NUM_OUT
+    script_pubkey = "6a4d" # OP_RETURN OP_PUSH2
+    script_pubkey += b2x(bytes_per_out.to_bytes(2, 'little')) # script_size bytes
+    for i in range(bytes_per_out):
+        script_pubkey += "01"
+
+    # Push fake a hash and height on the stack, followed by checkblockatheight
+    script_pubkey += "4c20"  # PUSH1 32bytes
+    script_pubkey += "0000000000000000000000000000000000000000000000000000000000000000" # 0
+    script_pubkey += "51"  # PUSH 1
+    script_pubkey += "b4" # OP_CHECKBLOCKATHEIGHT
+    ## concatenate NUM_OUT txouts of above script_pubkey which we'll insert before the txout for change
+    txouts = b2x(varlenEncode(NUM_OUT))
+    amount_str = b2x((amount_to_satoshi(amount) // NUM_OUT).to_bytes(8, 'little'))
+    for k in range(NUM_OUT):
         # add txout value
-        txouts += "b816000000000000"
+        txouts += amount_str
```

Comment:
A bit pedantic, so please double check if it's actually worth: should the remainder of the division `amount` by `NUM_OUT` be taken in consideration and being added to the last entry? If this doesn't conflict with fee/feerate just skip this comment.

Reply:
Done in the last update